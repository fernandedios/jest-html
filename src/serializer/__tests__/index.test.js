/* eslint-env jest */
import prettyFormat from 'pretty-format';

import * as serialize from '../';

describe('Serializer', () => {
  it('should correctly discard non-React components', () => {
    expect(serialize.test({})).toBeFalsy();
    expect(serialize.test(undefined)).toBeFalsy();
    expect(serialize.test(null)).toBeFalsy();
    expect(serialize.test(true)).toBeFalsy();
    expect(serialize.test(false)).toBeFalsy();
    expect(serialize.test('abc')).toBeFalsy();
  });

  it('should correctly detect React components', () => {
    const comp = {
      $$typeof: Symbol.for('react.test.json'),
    };
    expect(serialize.test(comp)).toBeTruthy();
  });

  it('should correctly serialize React components', () => {
    const obj = {
      $$typeof: Symbol.for('react.test.json'),
      type: 'div',
      props: {
        a: 3,
        b: 'foo',
        c: true,
        d: false,
        className: 'my-class',
        htmlFor: 'bar',
        style: {
          flexDirection: 'row',
          margin: 3,
          border: '1px solid blue',
          padding: 2,
          sth: undefined,
        },
      },
      children: ['Text', 5],
    };
    expect(prettyFormat(obj, { plugins: [serialize] })).toMatchSnapshot();
  });

  it('should correctly serialize Textareas', () => {
    const obj = {
      $$typeof: Symbol.for('react.test.json'),
      type: 'textarea',
      props: {
        value: 'One block\nof text\nwith accents: áéíóú',
        style: null,
      },
    };
    expect(prettyFormat(obj, { plugins: [serialize] })).toMatchSnapshot();
  });

  it('should correctly serialize childless components', () => {
    const obj = {
      $$typeof: Symbol.for('react.test.json'),
      type: 'div',
    };
    expect(prettyFormat(obj, { plugins: [serialize] })).toMatchSnapshot();
  });
});
