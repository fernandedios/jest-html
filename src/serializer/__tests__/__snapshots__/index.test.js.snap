exports[`Serializer should correctly serialize React components 1`] = `
"Object {
  \"$$typeof\": Symbol(react.test.json),
  \"__visited\": true,
  \"children\": Array [
    \"Text\",
    5,
  ],
  \"props\": Object {
    \"a\": 3,
    \"b\": \"foo\",
    \"c\": true,
    \"className\": \"my-class\",
    \"d\": false,
    \"htmlFor\": \"bar\",
    \"style\": Object {
      \"border\": \"1px solid blue\",
      \"flexDirection\": \"row\",
      \"margin\": 3,
      \"padding\": 2,
      \"sth\": undefined,
    },
  },
  \"type\": \"div\",
}
------------HTML PREVIEW---------------
<div
  b=\"foo\"
  c
  class=\"my-class\"
  for=\"bar\"
  style=\"border:1px solid blue;flex-direction:row;margin:3px;padding:2px;\"
>
  Text
  5
</div>"
`;

exports[`Serializer should correctly serialize Textareas 1`] = `
"Object {
  \"$$typeof\": Symbol(react.test.json),
  \"__visited\": true,
  \"props\": Object {
    \"style\": null,
    \"value\": \"One block
of text
with accents: áéíóú\",
  },
  \"type\": \"textarea\",
}
------------HTML PREVIEW---------------
<textarea value=\"One block
of text
with accents: áéíóú\">One block
of text
with accents: áéíóú</textarea>"
`;

exports[`Serializer should correctly serialize childless components 1`] = `
"Object {
  \"$$typeof\": Symbol(react.test.json),
  \"__visited\": true,
  \"type\": \"div\",
}
------------HTML PREVIEW---------------
<div></div>"
`;

exports[`Serializer should correctly serialize elements with one prop 1`] = `
"Object {
  \"$$typeof\": Symbol(react.test.json),
  \"__visited\": true,
  \"props\": Object {
    \"foo\": \"bar\",
  },
  \"type\": \"div\",
}
------------HTML PREVIEW---------------
<div foo=\"bar\"></div>"
`;

exports[`Serializer should correctly serialize elements with several props but only one valid one 1`] = `
"Object {
  \"$$typeof\": Symbol(react.test.json),
  \"__visited\": true,
  \"props\": Object {
    \"no\": false,
    \"yes\": true,
  },
  \"type\": \"div\",
}
------------HTML PREVIEW---------------
<div yes></div>"
`;

exports[`Serializer should correctly serialize elements with zero props 1`] = `
"Object {
  \"$$typeof\": Symbol(react.test.json),
  \"__visited\": true,
  \"type\": \"div\",
}
------------HTML PREVIEW---------------
<div></div>"
`;

exports[`Serializer should correctly serialize self-closing elements 1`] = `
"Object {
  \"$$typeof\": Symbol(react.test.json),
  \"__visited\": true,
  \"children\": Array [
    Object {
      \"type\": \"span\",
    },
    Object {
      \"type\": \"br\",
    },
    Object {
      \"props\": Object {
        \"src\": \"foo.png\",
      },
      \"type\": \"img\",
    },
    Object {
      \"type\": \"span\",
    },
  ],
  \"type\": \"div\",
}
------------HTML PREVIEW---------------
<div>
  <span></span>
  <br>
  <img src=\"foo.png\">
  <span></span>
</div>"
`;
